src/peer_class.cpp

```diff
@@ -87,7 +87,8 @@ namespace libtorrent
 		}
 		else
 		{
-			ret = m_peer_classes.size();
+			TORRENT_ASSERT(m_peer_classes.size() < 0x100);
+			ret = (peer_class_t)m_peer_classes.size();
```

Comment:
please use a C++ -style cast instead. I have enabled clang to warn about old-style casts